notifications-lambda/src/index.ts

```diff
@@ -47,45 +52,57 @@ export const handler = async (event: DynamoDBStreamEvent) => {
 
     if (userResults.Items) {
       await Promise.all(
-        userResults.Items.filter((user) => !!user.webPushSubscription)?.flatMap(
-          (user) =>
-            items
-              .filter(
-                // TODO: Include more scenarios that trigger desktop notification
-                (item) =>
-                  isUserMentioned(item, user as MyUser) ||
-                  doesUserManuallyHavePinboardOpen(item, user as MyUser)
-              )
-              .map((item) =>
-                webPush
-                  .sendNotification(
-                    user.webPushSubscription,
-                    JSON.stringify(item),
-                    {
-                      vapidDetails: {
-                        subject: "mailto:digitalcms.bugs@guardian.co.uk",
-                        publicKey: publicVapidKey,
-                        privateKey: privateVapidKey,
-                      },
-                    }
-                  )
-                  .then((result) => {
-                    if (result.statusCode < 300) {
-                      console.log(
-                        `Sent web push to ${user.email} with message ${item.message}`,
-                        result.body
-                      );
-                    } else {
-                      throw Error(result.body);
-                    }
-                  })
-                  .catch((errorPushing) => {
-                    console.error(
-                      `Failed to push to ${user.email} with message ${item.message}`,
-                      errorPushing
+        userResults.Items.reduce<UserWithWebPushSubscription[]>(
+          (acc, user) =>
+            user.webPushSubscription
+              ? [
+                  ...acc,
+                  {
+                    ...user,
+                    manuallyOpenedPinboardIds:
+                      user.manuallyOpenedPinboardIds?.values, // extract items from String Set
```

Comment:
this is the important bit, everything else is either indentation or type tightening etc

Reply:
does this come back as a js `Set`? (Tried reading the AWS docs which implies it is, but isn't clear). if so, values is a function https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/values

```suggestion
                      user.manuallyOpenedPinboardIds?.values(), // extract items from String Set
```

Reply:
no this comes back as some AWS construct, where `values` is the array. I've tested this PR, but feel free to deploy to CODE, to be sure. OR you can run it locally with `run.ts`